packages/block-library/src/button/style.scss

```diff
@@ -50,58 +49,35 @@ $blocks-block__margin: 0.5em;
 	}
 
 	&.wp-block-button__width-25 {
-		width: calc(25% - #{ $blocks-block__margin });
+		width: calc(25% - (var(--wp--style--block-gap, $blocks-block__margin) * 0.75));
 	}
 
 	&.wp-block-button__width-50 {
-		width: calc(50% - #{ $blocks-block__margin });
+		width: calc(50% - (var(--wp--style--block-gap, $blocks-block__margin) * 0.5));
 	}
 
 	&.wp-block-button__width-75 {
-		width: calc(75% - #{ $blocks-block__margin });
+		width: calc(75% - (var(--wp--style--block-gap, $blocks-block__margin) * 0.25));
 	}
 
 	&.wp-block-button__width-100 {
-		width: calc(100% - #{ $blocks-block__margin });
-
-		&:only-child {
-			margin-right: 0;
-			width: 100%;
-		}
+		width: 100%;
+		flex-basis: 100%;
 	}
 }
 
-// If the browser supports column-gap, use that instead of the margins above.
-@supports ( column-gap: #{ $blocks-block__margin } ) {
```

Comment:
Speaking of browser support, it's so nice to see the important ones green in [caniuse](https://caniuse.com/?search=gap). But I also recall something about Safari that was context for adding this check in the first place. Do you recall, @kjellr?